### PR TITLE
build(deps): bump react-native-fbsdk-next from 11.0.0 to 12.1.4

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-659] 

### Description

Bumps `react-native-fbsdk-next` from `11.0.0` to latest `12.1.4`

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-  bump react-native-fbsdk-next - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-659]: https://artsyproduct.atlassian.net/browse/PHIRE-659?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by gkartalis** [comment]
testing locally that everything works and will report back

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (bump react-native-fbsdk-next - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b1ca99f36e1dbae6069ecfe264144a4e2d95fb16
</p>

File: ios/Artsy/App/ARAppDelegate.mm [hunk 1]
```diff
@@ -129,6 +130,7 @@ - (void)setupForAppLaunch:(NSDictionary *)launchOptions
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
+  [FBSDKApplicationDelegate.sharedInstance initializeSDK];
```
Comment (gkartalis) [inline ios/Artsy/App/ARAppDelegate.mm:133]: > If you would like to initialize the Facebook SDK even earlier in startup for iOS, you need to include this code in your AppDelegate.m file now that auto-initialization is removed.

seems like we don't need this so I'm removing it

```suggestion
```

**Comment by gkartalis** [comment]
both android + iOS app flows seem to be working (but we are still not allowed to login due to the api)

**Review by MounirDhahri — APPROVED** [review]
Thanks!

**Pull request merged**